src/Features/LanguageServer/Protocol/Handler/RoslynRequestExecutionQueue.cs

```diff
@@ -0,0 +1,109 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.ErrorReporting;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Threading;
+using Roslyn.Utilities;
+using System.Collections.Immutable;
+using CommonLanguageServerProtocol.Framework;
+
+#nullable enable
+
+namespace Microsoft.CodeAnalysis.LanguageServer.Handler
+{
+    internal class RoslynRequestExecutionQueue : RequestExecutionQueue<RequestContext>
```

Comment:
Might be simpler for consumers if this was a service rather than used inheritance. ie, RequestExecutionQueue is concrete, but needs and IQueueItemFactoryService which roslyn and razor implement.

Reply:
Hmm, I think that makes sense to tell people what the important part that they're really overridding is. It also does a lot to convince me that y'all are right and `RequestContextType` ought to be an interface (which would allow us to set base-line demands such as lspServices, logger, clientCapabilities).